### PR TITLE
teams refactor: remove ChannelInfo

### PR DESCRIPTION
**Pull request by jakob223**
This PR takes ChannelInfo out of the store, and loads things when they're needed from the service instead.

I tested everything I touched except the journey cards, which I didn't know how to trigger (cc @mlsteele ) and the channel picker which isn't used outside the storybook yet.

cc @keybase/y2ksquad @mmaxim

### CODE REVIEW

**Comment by mlsteele** [comment]
You can trigger journeycards by creating chat channels `kb_cards_0_kb` through `kb_cards_7_kb`. Each channel will show [its card](https://github.com/keybase/client/blob/master/go/chat/journey_card_manager.go#L372:L372) all the time.

**Comment by zapu** [comment]
keybase cheat codes

File: shared/chat/conversation/input-area/normal/container.tsx [hunk 1]
```diff
@@ -61,50 +61,42 @@ let _channelSuggestions: Array<{channelname: string; teamname?: string}> = noCha
...
-  // partial list of channels that you have joined).
-  const convs = state.teams.teamIDToChannelInfos.get(teamID)
-  let suggestions: Array<{channelname: string}>
-  if (convs) {
-    suggestions = [...convs.values()].map(conv => ({
-      channelname: conv.channelname,
-    }))
-  } else {
-    suggestions = (state.chat2.inboxLayout?.bigTeams ?? []).reduce<Array<{channelname: string}>>((arr, t) => {
+  // TODO: get all the channels in the team, too, for this
```
Comment (buoyad) [inline shared/chat/conversation/input-area/normal/container.tsx]: Why not use the hook here for this?

File: shared/chat/conversation/messages/cards/team-journey/container.tsx [hunk 2]
```diff
@@ -45,7 +44,18 @@ const TeamJourneyContainer = (props: Props) => {
   let textComponent: React.ReactNode
   let image: Kb.IconType | null = null
   let actions: Array<Action> = []
-  let loadTeamID: TeamTypes.TeamID | undefined
+
+  // TODO: only call this in the event that it's one of the 2 cards that needs it
```
Comment (buoyad) [inline shared/chat/conversation/messages/cards/team-journey/container.tsx]: should we do this?

File: shared/mocks.js [hunk 3]
```diff
@@ -34,5 +35,7 @@ module.exports = {
     [/route-tree$/, '/../__mocks__/empty'],
     [/typed-connect/, '/../util/__mocks__/typed-connect'],
     [/util\/saga/, '/../__mocks__/saga'],
+    // TODO: this is not compatible with both mock places
```
Comment (buoyad) [inline shared/mocks.js]: ?

File: shared/teams/team/rows/index.tsx [hunk 4]
```diff
@@ -204,8 +209,11 @@ const makeRows = (
       break
     case 'channels': {
       rows.push({key: 'channel-header', type: 'channel-header'})
-      let channels: Array<{channel: Types.ChannelInfo; conversationIDKey: ChatTypes.ConversationIDKey}> = []
-      channelInfos?.forEach((channel, conversationIDKey) => channels.push({channel, conversationIDKey}))
+      let channels: Array<{
```
Comment (buoyad) [inline shared/teams/team/rows/index.tsx]: const

File: shared/actions/chat2/index.tsx [hunk 5]
```diff
@@ -2391,18 +2391,19 @@ function* loadSuggestionData(
...
   }
+  // TODO: do we keep the stuff in the store for this?
   // This only happens when user enters '#' which isn't that often. If this
   // becomes a problem, we can make a notification from service for when
   // channels change, and skip the load here if nothing has changed yet.
-  yield Saga.put(TeamsGen.createGetChannels({teamID}))
+  // yield Saga.put(TeamsGen.createGetChannels({teamID}))
 }
 
+// TODO: urgh what to do about this whole situation
```
[the remark this answers is not in the document]
Comment (jakob223) [reply shared/actions/chat2/index.tsx]: this action used to load the channels for all these teams, but I think it's fine if we only have channels you're in in the relevant team in the autocompleter.

File: shared/chat/conversation/input-area/normal/container.tsx (hunk 1 above)
Comment (jakob223) [reply shared/chat/conversation/input-area/normal/container.tsx]: You can only call hooks from within components or other hooks, and this is called from a container.

File: shared/mocks.js (hunk 3 above)
Comment (jakob223) [reply shared/mocks.js]: webpack.config.js specifies the locations of the mocks with respect to webpack.config.js, whereas metro.config.js specifies the locations of the mocks with respect to the files themselves. Will fix.

File: shared/chat/conversation/input-area/normal/container.tsx (hunk 1 above)
Comment (buoyad) [reply shared/chat/conversation/input-area/normal/container.tsx]: Ah mhmm

**Comment by buoyad** [comment]
There are so many places where these channel suggestions for chat are handled.

File: shared/chat/conversation/input-area/normal/container.tsx (hunk 1 above)
Comment (buoyad) [reply shared/chat/conversation/input-area/normal/container.tsx]: From what I can tell this is status quo except that channel suggestions in a team only show the channels you're in. We should ticket / fix if possible.

**Pull request merged**